tests/experimental/pydantic/test_conversion.py

```diff
@@ -787,22 +787,6 @@ def test_convert_input_types_to_pydantic_default_and_default_factory():
         class User(pydantic.BaseModel):
             password: Optional[str] = Field(default=None, default_factory=lambda: None)
 
-    # If the user defines both through a hacky way, we'll still going to catch it
-    hacked_field = Field(default_factory=lambda: None)
-
-    class User2(pydantic.BaseModel):
-        password: Optional[str] = hacked_field
-
-    hacked_field.default = None
-    with pytest.raises(
-        BothDefaultAndDefaultFactoryDefinedError,
-        match=("Not allowed to specify both default and default_factory."),
-    ):
-
-        @strawberry.experimental.pydantic.input(User2)
-        class UserInput:
-            password: strawberry.auto
-
```

Comment:
turns out hacking it like this wasn't possible. 
doing
```
hacked_field.default = None 
```
didn't actually change the default. pydantic under the hood would ignore the change somehow.

the test only worked because it already was None by default